File: preliminary_analysis/baseline_cpp_results/instruct_generations/cpp/files_before_feedback/question-79-pass-1.cpp
#include <bits/stdc++.h> 
 using namespace std ;
 
bool isConcentrichexagonal ( int N ) {
    // Check if N is a perfect square
    int sqrtN = (int)sqrt(N);
    if (sqrtN * sqrtN!= N) return false;

    // Check if N is a hexagonal number
    int x = N * (2 * N - 1);
    int y = N * (3 * N - 1);
    if (x % 2 == 0 && y % 2 == 0) return true;

    return false;
}

File: fine_tuning/results/test_orig_files/question-79-pass-1.py
import math



def isConcentrichexagonal(N):
    # Check if N is a perfect square
    if math.sqrt(N) % 1 == 0:
        # Get the square root of N
        square_root = int(math.sqrt(N))
        # Check if the number of edges in the hexagon is equal to the number of edges in the regular hexagon with the same side length
        if len(hexagon_edges(square_root)) == len(hexagon_edges(N)):
            return True
    return False

def hexagon_edges(N):
    # Return a list of the edges of the hexagon with the given side length
    return [N * math.cos(math.radians(60)) + N * math.cos(math.radians(120)), N * math.cos(math.radians(60)) - N * math.cos(math.radians(120)), -N * math.cos(math.radians(60)) + N * math.cos(math.radians(120)), -N * math.cos(math.radians(60)) - N * math.cos(math.radians(120))]
